Linux/s2e-kprobe/s2e-kprobe.c

```diff
@@ -0,0 +1,741 @@
+/// copies of the Software, and to permit persons to whom the Software is
+/// furnished to do so, subject to the following conditions:
+///
+/// The above copyright notice and this permission notice shall be included in all
+/// copies or substantial portions of the Software.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+/// SOFTWARE.
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/sched.h>
+#include <linux/kprobes.h>
+#include <linux/slab.h>
+#include <linux/binfmts.h>
+#include <asm/processor.h>
+#include <linux/dcache.h>
+#include <linux/namei.h>
+#include <linux/mman.h>
+#include <linux/fs.h>
+#include <linux/kasan.h>
+#include <linux/printk.h>
+#include <linux/version.h>
+
+#include <linux/preempt.h>
+
+#include <linux/mm_types.h>
+
+#include <linux/elf.h>
+#include <linux/elf-randomize.h>
+
+#include <linux/sched/task_stack.h>
+
+#include <s2e/s2e.h>
+#include <s2e/linux/linux_monitor.h>
+
+#include "s2e-kprobe.h"
+
+#ifdef S2E_ENABLED
+#define print(fmt, arg...) s2e_printf(fmt, ##arg)
+//#define print(fmt, arg...) printk(KERN_INFO fmt, ##arg)
+#else
+#define print(fmt, arg...) printk(KERN_INFO fmt, ##arg)
+#endif //S2E_ENABLED
+
+struct task_struct *s2e_current;
+struct nameidata;
+
+/* mprotect */
+static int handler_mprotect_entry(struct kretprobe_instance *p, struct pt_regs *regs) {
+    struct mprotect_data *data = (struct mprotect_data *) p->data;
+    data->start = regs->di;
+    data->len = regs->si;
+    data->prot = regs->dx;
+
+#ifdef S2E_DEBUG
+    print("%d) mprotect entry: %lx, flags = 0x%lx start = 0x%lx, len = %lx, prot = 0x%lx\n", current->pid, regs->ip, regs->flags, data->start, data->len, data->prot);
+#endif
+    return 0;
+}
+
+static int handler_mprotect_return(struct kretprobe_instance *p, struct pt_regs *regs) {
+    unsigned long retval = regs_return_value(regs);
+    struct mprotect_data *data = (struct mprotect_data *) p->data;
+    if (!retval) {
+#ifdef S2E_ENABLED
+        s2e_linux_mprotect(current, data->start, data->len, data->prot);
+#endif
+    }
+#ifdef S2E_DEBUG
+    print("%d) mprotect return: start = 0x%lx, len = %lx, prot = 0x%lx, retval = %lx\n", current->pid, data->start, data->len, data->prot, retval);
+#endif
+    return 0;
+}
+
+
+/* panic */
+//TODO panic needs some fixing
+static int handler_panic_entry(struct kprobe *p, struct pt_regs *regs) {
+    const char *fmt = (const char *)regs->di;
+    static char buf[1024];
+    va_list args;
+    //va_start(args, fmt);
```

Comment:
Can you please clean up the code, e.g., remove commented-out code, run clang-format, etc.